catroid/src/test/java/org/catrobat/catroid/test/content/actions/WebRequestActionTest.java

```diff
@@ -185,4 +190,50 @@ public void testCancelledCallAndResendRequest() {
 		verify(StageActivity.stageListener.webConnectionHolder, times(2)).addConnection(any());
 		verify(webConnection, times(2)).sendWebRequest();
 	}
+
+	@Test
+	public void testSuccessfulResponseWithInputVariable() {
+		UserVariable inputVariable = new UserVariable(TEST_INPUTVARIABLE, TEST_URL);
+		FormulaElement formulaElement = new FormulaElement(
+				FormulaElement.ElementType.USER_VARIABLE, inputVariable.getName(), null);
+		final String responseString = "Response";
+
+		Formula formula = Mockito.spy(new Formula(formulaElement));
+
+		testSprite.addUserVariable(userVariable);
+		testSprite.addUserVariable(inputVariable);
```

Comment:
Could you split up the test in a way that its sepparated from setup and the code lines which really do test something. Also please think about implementing the teardown function because you are using static members like the `StageActivity.stageListener. ...`. This mocks will remain for further tests and may bring them to fail if not reset

Reply:
Setup is now in a seperate function.

Can you please specify, how the existing teardown-function is not sufficient?
I have only seen other tests setting the **Stagelister.stageListener** to null.